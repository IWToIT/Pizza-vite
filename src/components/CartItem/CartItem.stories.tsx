import type { Meta, StoryObj } from "@storybook/react";
import { CartItem } from "./index";

const meta: Meta<typeof CartItem> = {
  component: CartItem,
  args: {
    id: "1",
    title: "Pizza",
    type: "Vegetarian",
    size: 12,
    price: 10,
    count: 2,
    imageUrl: "pizza.jpg",
  },
};

export default meta;
type Story = StoryObj<typeof CartItem>;

export const Primary: Story = {};
