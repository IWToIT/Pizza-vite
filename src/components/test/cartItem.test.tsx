import React from 'react';
import { render } from './test-utils';
import { screen } from '@testing-library/react';
import { CartItem } from '../CartItem';

test('is the cart block displayed with all the properties', () => {
  const props = {
    id: '1',
    title: 'Pizza',
    type: 'Vegetarian',
    size: 12,
    price: 10,
    count: 2,
    imageUrl: 'pizza.jpg',
  };

  render(<CartItem {...props} />);
  expect(screen.getByTestId('cartItem-element')).toBeInTheDocument();
  expect(screen.getByTestId('cartItem-element')).toMatchSnapshot();
  expect(screen.getByAltText('Pizza')).toBeInTheDocument();
  expect(screen.getByText('Pizza')).toBeInTheDocument();
  expect(screen.getByText('Vegetarian, 12 см.')).toBeInTheDocument();
  expect(screen.getByText('2')).toBeInTheDocument();
  expect(screen.getByText('20 руб.')).toBeInTheDocument();
});
