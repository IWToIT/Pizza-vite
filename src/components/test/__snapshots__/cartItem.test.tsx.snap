// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is the cart block displayed with all the properties 1`] = `
<div
  class="cart__item"
  data-testid="cartItem-element"
>
  <div
    class="cart__item-img"
  >
    <img
      alt="Pizza"
      class="pizza-block__image"
      src="pizza.jpg"
    />
  </div>
  <div
    class="cart__item-info"
  >
    <h3>
      Pizza
    </h3>
    <p>
      Vegetarian
      , 
      12
       см.
    </p>
  </div>
  <div
    class="cart__item-count"
  >
    <div
      class="button button--outline button--circle cart__item-count-minus"
    >
      <svg
        fill="none"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.92001 3.84V5.76V8.64C5.92001 9.17016 5.49017 9.6 4.96001 9.6C4.42985 9.6 4.00001 9.17016 4.00001 8.64L4 5.76L4.00001 3.84V0.96C4.00001 0.42984 4.42985 0 4.96001 0C5.49017 0 5.92001 0.42984 5.92001 0.96V3.84Z"
          fill="#EB5A1E"
        />
        <path
          d="M5.75998 5.92001L3.83998 5.92001L0.959977 5.92001C0.429817 5.92001 -2.29533e-05 5.49017 -2.29301e-05 4.96001C-2.2907e-05 4.42985 0.429817 4.00001 0.959977 4.00001L3.83998 4L5.75998 4.00001L8.63998 4.00001C9.17014 4.00001 9.59998 4.42985 9.59998 4.96001C9.59998 5.49017 9.17014 5.92001 8.63998 5.92001L5.75998 5.92001Z"
          fill="#EB5A1E"
        />
      </svg>
    </div>
    <b>
      2
    </b>
    <div
      class="button button--outline button--circle cart__item-count-plus"
    >
      <svg
        fill="none"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.92001 3.84V5.76V8.64C5.92001 9.17016 5.49017 9.6 4.96001 9.6C4.42985 9.6 4.00001 9.17016 4.00001 8.64L4 5.76L4.00001 3.84V0.96C4.00001 0.42984 4.42985 0 4.96001 0C5.49017 0 5.92001 0.42984 5.92001 0.96V3.84Z"
          fill="#EB5A1E"
        />
        <path
          d="M5.75998 5.92001L3.83998 5.92001L0.959977 5.92001C0.429817 5.92001 -2.29533e-05 5.49017 -2.29301e-05 4.96001C-2.2907e-05 4.42985 0.429817 4.00001 0.959977 4.00001L3.83998 4L5.75998 4.00001L8.63998 4.00001C9.17014 4.00001 9.59998 4.42985 9.59998 4.96001C9.59998 5.49017 9.17014 5.92001 8.63998 5.92001L5.75998 5.92001Z"
          fill="#EB5A1E"
        />
      </svg>
    </div>
  </div>
  <div
    class="cart__item-price"
  >
    <b>
      20
       руб.
    </b>
  </div>
  <div
    class="cart__item-remove"
  >
    <div
      class="button button--outline button--circle"
    >
      <svg
        fill="none"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.92001 3.84V5.76V8.64C5.92001 9.17016 5.49017 9.6 4.96001 9.6C4.42985 9.6 4.00001 9.17016 4.00001 8.64L4 5.76L4.00001 3.84V0.96C4.00001 0.42984 4.42985 0 4.96001 0C5.49017 0 5.92001 0.42984 5.92001 0.96V3.84Z"
          fill="#EB5A1E"
        />
        <path
          d="M5.75998 5.92001L3.83998 5.92001L0.959977 5.92001C0.429817 5.92001 -2.29533e-05 5.49017 -2.29301e-05 4.96001C-2.2907e-05 4.42985 0.429817 4.00001 0.959977 4.00001L3.83998 4L5.75998 4.00001L8.63998 4.00001C9.17014 4.00001 9.59998 4.42985 9.59998 4.96001C9.59998 5.49017 9.17014 5.92001 8.63998 5.92001L5.75998 5.92001Z"
          fill="#EB5A1E"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should render the cart item with the correct details 1`] = `
<div
  class="cart__item"
  data-testid="cart-element"
>
  <div
    class="cart__item-img"
  >
    <img
      alt="Pizza"
      class="pizza-block__image"
      src="pizza.jpg"
    />
  </div>
  <div
    class="cart__item-info"
  >
    <h3>
      Pizza
    </h3>
    <p>
      Vegetarian
      , 
      12
       см.
    </p>
  </div>
  <div
    class="cart__item-count"
  >
    <div
      class="button button--outline button--circle cart__item-count-minus"
    >
      <svg
        fill="none"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.92001 3.84V5.76V8.64C5.92001 9.17016 5.49017 9.6 4.96001 9.6C4.42985 9.6 4.00001 9.17016 4.00001 8.64L4 5.76L4.00001 3.84V0.96C4.00001 0.42984 4.42985 0 4.96001 0C5.49017 0 5.92001 0.42984 5.92001 0.96V3.84Z"
          fill="#EB5A1E"
        />
        <path
          d="M5.75998 5.92001L3.83998 5.92001L0.959977 5.92001C0.429817 5.92001 -2.29533e-05 5.49017 -2.29301e-05 4.96001C-2.2907e-05 4.42985 0.429817 4.00001 0.959977 4.00001L3.83998 4L5.75998 4.00001L8.63998 4.00001C9.17014 4.00001 9.59998 4.42985 9.59998 4.96001C9.59998 5.49017 9.17014 5.92001 8.63998 5.92001L5.75998 5.92001Z"
          fill="#EB5A1E"
        />
      </svg>
    </div>
    <b>
      2
    </b>
    <div
      class="button button--outline button--circle cart__item-count-plus"
    >
      <svg
        fill="none"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.92001 3.84V5.76V8.64C5.92001 9.17016 5.49017 9.6 4.96001 9.6C4.42985 9.6 4.00001 9.17016 4.00001 8.64L4 5.76L4.00001 3.84V0.96C4.00001 0.42984 4.42985 0 4.96001 0C5.49017 0 5.92001 0.42984 5.92001 0.96V3.84Z"
          fill="#EB5A1E"
        />
        <path
          d="M5.75998 5.92001L3.83998 5.92001L0.959977 5.92001C0.429817 5.92001 -2.29533e-05 5.49017 -2.29301e-05 4.96001C-2.2907e-05 4.42985 0.429817 4.00001 0.959977 4.00001L3.83998 4L5.75998 4.00001L8.63998 4.00001C9.17014 4.00001 9.59998 4.42985 9.59998 4.96001C9.59998 5.49017 9.17014 5.92001 8.63998 5.92001L5.75998 5.92001Z"
          fill="#EB5A1E"
        />
      </svg>
    </div>
  </div>
  <div
    class="cart__item-price"
  >
    <b>
      20
       руб.
    </b>
  </div>
  <div
    class="cart__item-remove"
  >
    <div
      class="button button--outline button--circle"
    >
      <svg
        fill="none"
        height="10"
        viewBox="0 0 10 10"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.92001 3.84V5.76V8.64C5.92001 9.17016 5.49017 9.6 4.96001 9.6C4.42985 9.6 4.00001 9.17016 4.00001 8.64L4 5.76L4.00001 3.84V0.96C4.00001 0.42984 4.42985 0 4.96001 0C5.49017 0 5.92001 0.42984 5.92001 0.96V3.84Z"
          fill="#EB5A1E"
        />
        <path
          d="M5.75998 5.92001L3.83998 5.92001L0.959977 5.92001C0.429817 5.92001 -2.29533e-05 5.49017 -2.29301e-05 4.96001C-2.2907e-05 4.42985 0.429817 4.00001 0.959977 4.00001L3.83998 4L5.75998 4.00001L8.63998 4.00001C9.17014 4.00001 9.59998 4.42985 9.59998 4.96001C9.59998 5.49017 9.17014 5.92001 8.63998 5.92001L5.75998 5.92001Z"
          fill="#EB5A1E"
        />
      </svg>
    </div>
  </div>
</div>
`;
